compendium/DeclarativeServices/test/gtest/TestAsyncWorkService.cpp

```diff
@@ -471,4 +421,50 @@ namespace test
         fut.get();
     }
 
+    /**
+     * Verify that a service's configuration can be updated from with a 'Modified' callback without
+     * deadlocking the framework trying to get the lock in configNotifier
+     */
+    TEST_F(TestAsyncWorkServiceEndToEnd, testUpdateConfigFromWithinModifiedCallback)
+    {
+        auto const& param = std::make_shared<AsyncWorkServiceThreadPool>(10);
+        auto context = framework.GetBundleContext();
+
+        auto reg = context.RegisterService<cppmicroservices::async::AsyncWorkService>(param);
+        auto sr = context.GetServiceReference<cppmicroservices::async::AsyncWorkService>();
+        auto asyncWorkService = context.GetService<cppmicroservices::async::AsyncWorkService>(sr);
+
+        US_UNUSED(asyncWorkService);
+        // Start bundle
+        std::string componentName = "sample::ServiceComponentCA10";
+        auto testBundle = ::test::InstallAndStartBundle(context, "TestBundleDSCA10");
+        ::test::InstallAndStartBundle(context, "TestBundleDSCA03");
+        ::test::InstallAndStartConfigAdmin(context);
+
+        // Get a service reference to ConfigAdmin.
+        auto sr1 = context.GetServiceReference<cppmicroservices::service::cm::ConfigurationAdmin>();
+        auto configAdminService = context.GetService<cppmicroservices::service::cm::ConfigurationAdmin>(sr1);
+        ASSERT_TRUE(configAdminService) << "GetService failed for ConfigurationAdmin";
+
+        auto configObject = configAdminService->GetConfiguration(componentName);
+
+        cppmicroservices::AnyMap props(cppmicroservices::AnyMap::UNORDERED_MAP_CASEINSENSITIVE_KEYS);
```

Comment:
Could use `configObject->GetProperties()` to simplify this?

Reply:
Probably :)